cli/tools/test.rs

```diff
@@ -476,12 +478,34 @@ async fn test_specifier(
 
   // We only execute the specifier as a module if it is tagged with TestMode::Module or
   // TestMode::Both.
+  let mut await_cjs_modules = false;
   if mode != TestMode::Documentation {
-    // We execute the module module as a side module so that import.meta.main is not set.
-    worker.execute_side_module(&specifier).await?;
+    if compat_mode {
+      worker.execute_side_module(&compat::GLOBAL_URL).await?;
+      worker.execute_side_module(&compat::MODULE_URL).await?;
+
+      let use_esm_loader = compat::check_if_should_use_esm_loader(&specifier)?;
+
+      if use_esm_loader {
+        worker.execute_side_module(&specifier).await?;
+      } else {
+        await_cjs_modules = true;
+        compat::load_cjs_module(
+          &mut worker.js_runtime,
+          &specifier.to_file_path().unwrap().display().to_string(),
+          false,
+        )?;
+      }
+    } else {
+      // We execute the module module as a side module so that import.meta.main is not set.
+      worker.execute_side_module(&specifier).await?;
+    }
   }
 
   worker.dispatch_load_event(&located_script_name!())?;
+  if await_cjs_modules {
+    worker.run_event_loop(false).await?;
+  }
```

Comment:
Since CJS modules execute synchronously I think this could be moved after `load_cjs_module`

Reply:
And then `await_cjs_modules` could be removed.

Reply:
Done